superset-frontend/src/SqlLab/components/QuerySearch.tsx

```diff
@@ -0,0 +1,283 @@
+  height: 100%;
+`;
+
+const TableStyles = styled.div`
+  table {
+    background-color: ${({ theme }) => theme.colors.grayscale.light4};
+  }
+
+  .table > thead > tr > th {
+    border-bottom: ${({ theme }) => theme.gridUnit / 2}px solid
+      ${({ theme }) => theme.colors.grayscale.light2};
+    background: ${({ theme }) => theme.colors.grayscale.light4};
+  }
+`;
+
+const StyledTableStylesContainer = styled.div`
+  overflow: auto;
+`;
+function QuerySearch({ actions, displayLimit }: propTypes) {
+  const [databaseId, setDatabaseId] = useState<any>('');
+  const [userId, setUserId] = useState<string>('');
+  const [searchText, setSearchText] = useState<string>('');
+  const [from, setFrom] = useState<string>('28 days ago');
+  const [to, setTo] = useState<string>('now');
+  const [status, setStatus] = useState<string>('success');
+  const [queriesArray, setQueriesArray] = useState<any>([]);
+  const [queriesLoading, setQueriesLoading] = useState<boolean>(true);
+
+  const getTimeFromSelection = (selection: string) => {
+    switch (selection) {
+      case 'now':
+        return now();
+      case '1 hour ago':
+        return epochTimeXHoursAgo(1);
+      case '1 day ago':
+        return epochTimeXDaysAgo(1);
+      case '7 days ago':
+        return epochTimeXDaysAgo(7);
+      case '28 days ago':
+        return epochTimeXDaysAgo(28);
+      case '90 days ago':
+        return epochTimeXDaysAgo(90);
+      case '1 year ago':
+        return epochTimeXYearsAgo(1);
+      default:
+        return null;
+    }
+  };
+
+  const insertParams = (baseUrl: string, params: string[]) => {
+    const validParams = params.filter(function (p) {
+      return p !== '';
+    });
+    return `${baseUrl}?${validParams.join('&')}`;
+  };
+
+  const refreshQueries = async () => {
+    setQueriesLoading(true);
+    const params = [
+      userId && `user_id=${userId}`,
+      databaseId && `database_id=${databaseId}`,
+      searchText && `search_text=${searchText}`,
+      status && `status=${status}`,
+      from && `from=${getTimeFromSelection(from)}`,
+      to && `to=${getTimeFromSelection(to)}`,
+    ];
+
+    // make into async method
+    try {
+      const promise = await SupersetClient.get({
+        endpoint: insertParams('/superset/search_queries', params),
+      });
+      setQueriesArray(promise.json);
+    } catch (err) {
+      actions.addDangerToast(t('An error occurred when refreshing queries'));
+    } finally {
+      setQueriesLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    refreshQueries();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onUserClicked = (userId: string) => {
+    setUserId(userId);
+    refreshQueries();
+  };
+
+  const onDbClicked = (dbId: string) => {
+    setDatabaseId(dbId);
+    refreshQueries();
+  };
+
+  const onKeyDown = (event: any) => {
+    if (event.keyCode === 13) {
+      refreshQueries();
+    }
+  };
+
+  const onChange = (e: any) => {
+    e.persist();
```

Comment:
what does this do?

Reply:
oh! Yeah this is totally because of the debouncer. Apparently when you have a debouncer in hooks the event.target.value gets lost, so you have to tell it to persist the value. Without this formatting you could type something into the search bar, and it would show up on the screen but it wouldn't actually search for it. Kind of wild.